contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java

```diff
@@ -230,6 +239,18 @@ private void openFile(FileScanFramework.FileSchemaNegotiator negotiator) {
     try {
       fsStream = negotiator.fileSystem().openPossiblyCompressedStream(split.getPath());
 
+      if (readerConfig.maxArraySize >= 0) {
```

Comment:
Is it possible to add the unit tests for these three options? we can refer to the : `TestExcelFormat#testLastRowQuery()`.